user_tools/src/spark_rapids_tools/utils/util.py

```diff
@@ -248,10 +248,21 @@ def get_system_memory_in_gb(cls) -> int:
         ps_memory = psutil.virtual_memory()
         return int(0.8 * ps_memory.total / (1024 ** 3))
 
+    @classmethod
+    def get_max_jvm_threads(cls) -> int:
+        """
+        Get the total cpu_count.
+        """
+        # Maximum number of threads that can be used in the tools JVM.
+        # cpu_count returns the logical number of cores. So, we take a 50% to get better representation
+        # of physical cores.
+        return min(6, (psutil.cpu_count() + 1) // 2)
```

Comment:
Can we use `psutil.cpu_count(logical=False)` to get number of physical cores?

Reply:
Very good question.
I actually tried that out on my local development and I got the same return value.
All the docs suggest that it should be different result.
I did not dive deeper to see why that's the case, but my intuition is that it could be a kernel (OS) or a library compatibility thing.
That's why I decided to lower down the value returned by dividing by 2. IIRC, the core tools code comments said that we use number_cores/4 as default num_threads

Reply:
I see. I think in our Mac we have same number of physical and logical cores. In linux machines these should be different. Although it should not matter much.